### PR TITLE
Halide recipe that gets 64% derivability.

### PR DESCRIPTION
**Pull request by vee-effekt**
Also, changes to the logger so you can choose not to attempt derivability.

### CODE REVIEW

File: src/enumo/scheduler.rs [hunk 1]
```diff
@@ -137,7 +137,7 @@ impl Scheduler {
         egraph: &EGraph<L, SynthAnalysis>,
         ruleset: &Ruleset<L>,
     ) -> EGraph<L, SynthAnalysis> {
-        self.run_internal(egraph, ruleset, None, 200_000)
```
Comment (ajpal) [inline src/enumo/scheduler.rs:140]: why did you change this back? @oflatt  lowered it in https://github.com/uwplse/ruler/pull/175

File: src/logger.rs [hunk 2]
```diff
@@ -112,33 +113,47 @@ pub fn get_derivability_results<L: SynthLanguage>(
...
-    });
-
-    (
-        (can_f.len(), can_b.len()),
-        (time_f, time_b),
-        derivability_results,
-    )
+        let limits = if let DeriveType::AllRules = derive_type {
+            Limits {
+                iter: 2,
```
Comment (ajpal) [inline src/logger.rs:132]: since we're not doing derive for halide on nightly anymore, can we put this back at 5?
or are we getting rid of All entirely?

File: tests/recipes/halide.rs [hunk 3]
```diff
@@ -70,6 +61,53 @@ pub fn halide_rules() -> Ruleset<Pred> {
         ),
         all_rules.clone(),
     );
-    all_rules.extend(full);
+    println!("full finished.");
+    all_rules.to_file("thru_full.rules");
```
Comment (ajpal) [inline tests/recipes/halide.rs:65]: take out these write to files

File: tests/halide.rs [hunk 4]
```diff
@@ -322,12 +319,15 @@ mod test {
...
 
+        logger::write_output(&all_rules, &baseline, "halide", "halide", duration, false);
+
         logger::write_output(
             &oopsla_halide,
             &baseline,
             "oopsla halide (1 iter)",
             "halide",
             oopsla_duration,
+            true,
```
Comment (ajpal) [inline tests/halide.rs:330]: make this false too

File: src/logger.rs (hunk 2 above)
Comment (vee-effekt) [reply src/logger.rs:132]: I think we're getting rid of it entirely.

File: src/enumo/scheduler.rs (hunk 1 above)
Comment (vee-effekt) [reply src/enumo/scheduler.rs:140]: I'll put it back, wanted to see what would happen.

File: tests/halide.rs (hunk 4 above)
Comment (vee-effekt) [reply tests/halide.rs:330]: oh, why?

File: src/logger.rs [hunk 5]
```diff
@@ -112,33 +113,47 @@ pub fn get_derivability_results<L: SynthLanguage>(
...
     derive_type: DeriveType,
     baseline: &Ruleset<L>,
+    derive: bool,
 ) -> ((usize, usize), (Duration, Duration), Value) {
-    let limits = if let DeriveType::AllRules = derive_type {
-        Limits {
-            iter: 2,
-            node: 100_000,
-        }
+    if !derive {
```
Comment (ajpal) [inline src/logger.rs:118]: I think it's weird/confusing to have this check in this function. it would make more sense to have it in `write_output`. Like, we shouldn't be calling a function called `get_derivability_results` if we don't want to actually run derivability

File: tests/halide.rs (hunk 4 above)
Comment (ajpal) [reply tests/halide.rs:330]: if we're not running derivability for the enumo halide recipe, there's no point running it for the baseline. i think it'll make it more clear in the nightly report that it just didn't run if everything is 0s

File: src/logger.rs (hunk 5 above)
Comment (vee-effekt) [reply src/logger.rs:118]: it's a lot easier to write this way

Comment (vee-effekt) [reply src/logger.rs:148]: but i'll change it
